### ISSUE TITLE
Rationale for Varying Epsilon 

### ISSUE THREAD

**Issue by 0xJchen** [gouxiangchen/dueling-DQN-pytorch#3]
Thanks for the great work. One small question:
Is there any reference for the varying epsilon along training?
```
if epsilon > FINAL_EPSILON:
    epsilon -= (INITIAL_EPSILON - FINAL_EPSILON) / EXPLORE
```

**Comment by gouxiangchen** [comment]
Hi,

I am sorry for the late reply.

It is mentioned in the DQN paper ["Human-level control through deep reinforcement learning"](https://www.nature.com/articles/nature14236/),

**The behaviour policy during training was <a href="https://www.codecogs.com/eqnedit.php?latex=\varepsilon" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\varepsilon" title="\varepsilon" /></a>-greedy with <a href="https://www.codecogs.com/eqnedit.php?latex=\varepsilon" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\varepsilon" title="\varepsilon" /></a> annealed linearly from 1.0 to 0.1 over the first million frames, and fixed at 0.1 thereafter.**

In the early training, the agent knows nothing with the environments and should explore more with higher <a href="https://www.codecogs.com/eqnedit.php?latex=\varepsilon" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\varepsilon" title="\varepsilon" /></a>. While the capability improves, <a href="https://www.codecogs.com/eqnedit.php?latex=\varepsilon" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\varepsilon" title="\varepsilon" /></a> decrease accordingly.

May this reply helps :).

**Comment by 0xJchen** [comment]
Got it. Thanks!
